src/drivers/uavcannode/Subscribers/MovingBaselineData.hpp

```diff
@@ -72,23 +72,34 @@ class RTCMStream :
 	void PrintInfo() const override
 	{
 		printf("\t%s:%d -> %s\n",
-		       uavcan::equipment::gnss::RTCMStream::getDataTypeFullName(),
-		       uavcan::equipment::gnss::RTCMStream::DefaultDataTypeID,
+		       ardupilot::gnss::MovingBaselineData::getDataTypeFullName(),
+		       ardupilot::gnss::MovingBaselineData::DefaultDataTypeID,
 		       _gps_inject_data_pub.get_topic()->o_name);
 	}
 
 private:
-	void callback(const uavcan::ReceivedDataStructure<uavcan::equipment::gnss::RTCMStream> &msg)
+	void callback(const uavcan::ReceivedDataStructure<ardupilot::gnss::MovingBaselineData> &msg)
 	{
-		gps_inject_data_s gps_inject_data{};
+		// Don't republish a message from ourselves
+		if (msg.getSrcNodeID().get() != getNode().getNodeID().get()) {
+			gps_inject_data_s gps_inject_data{};
 
-		gps_inject_data.len = msg.data.size();
+			//static_assert(sizeof(gps_inject_data.data) > sizeof(msg.data), "Buffer size too small");
+			if (sizeof(gps_inject_data.data) > sizeof(msg.data)) {
+				PX4_ERR("Buffer size too small. gps_inject_data.data = %d > msg.data = %d", sizeof(gps_inject_data.data),
+					sizeof(msg.data));
+			}
 
-		//gps_inject_data.flags = gps_rtcm_data_msg.flags;
-		memcpy(gps_inject_data.data, &msg.data[0], gps_inject_data.len);
+			gps_inject_data.len = msg.data.size();
 
-		gps_inject_data.timestamp = hrt_absolute_time();
-		_gps_inject_data_pub.publish(gps_inject_data);
+			memcpy(gps_inject_data.data, &msg.data[0], gps_inject_data.len);
+
+			gps_inject_data.timestamp = hrt_absolute_time();
+
+			gps_inject_data.device_id = msg.getSrcNodeID().get();
```

Comment:
A valid device id would have the bus type set to UAVCAN, the bus, and the node id as the address. https://github.com/PX4/PX4-Autopilot/blob/fb54324f5612d5565cfbf366d295799a0641dd22/src/lib/drivers/device/Device.hpp#L160-L165

For the device type you can use the generic DRV_GPS_DEVTYPE_UAVCAN as unfortunately we don't know the specific type over CAN.

https://github.com/PX4/PX4-Autopilot/blob/fb54324f5612d5565cfbf366d295799a0641dd22/src/drivers/drv_sensor.h#L161